src/host_capabilities/kubernetes.rs

```diff
@@ -124,24 +124,84 @@ where
     })
 }
 
-/// Describe the set of parameters used by the `can_i` function.
-#[derive(Serialize, Deserialize, Debug)]
+impl From<SubjectAccessReviewRequest> for SubjectAccessReviewSpec {
+    fn from(request: SubjectAccessReviewRequest) -> Self {
+        SubjectAccessReviewSpec {
+            user: Some(request.user),
+            groups: request.groups,
+            resource_attributes: Some(request.resource_attributes.into()),
+            ..Default::default()
+        }
+    }
+}
+
+impl From<ResourceAttributes> for k8s_openapi::api::authorization::v1::ResourceAttributes {
+    fn from(attrs: ResourceAttributes) -> Self {
+        k8s_openapi::api::authorization::v1::ResourceAttributes {
+            namespace: attrs.namespace,
+            verb: Some(attrs.verb),
+            group: attrs.group,
+            resource: Some(attrs.resource),
+            subresource: attrs.subresource,
+            name: attrs.name,
+            version: attrs.version,
+            ..Default::default()
+        }
+    }
+}
+
+/// Describe the set of parameters used by the `can_i` function. The values in
+/// this struct will be used to build the SubjectAccessReview resources sent to
+/// the Kubernetes API to verify if the user is allowed to perform some operation
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Default, Hash, Clone)]
 pub struct SubjectAccessReviewRequest {
```

Comment:
If this is the set of parameters used by the can_i function, I would expect this struct to be named `CanIRequest{}` (following the pattern of `GetResourceRequest{}`, `ListAllResourcesRequest{}`, and so). Renaming also helps the confusion on having 2 `disable_cache` fields, one in this current  `SubjectAccessReviewRequest{}` and another in the policy-evaluator's `KubernetesCanI{}`.

I would expect something like the following:

```
pub struct CanIRequest{
  request: SubjectAccessReviewRequest, // gets used to do .into()
  disable_cache: bool,
}

pub struct SubjectAccessReviewRequest {
  pub groups: Option<Vec<String>>,
  pub user: String,
  pub resource_attributes: ResourceAttributes,
}
```